### ISSUE TITLE
Errors not always caught

### ISSUE THREAD

**Issue by longouyang** [probmods/webppl-editor#1]
The ReferenceError is successfully caught here:

``` js
print('start 2');
meh
hist(repeat(1e5, flip))
print('end 2')
```

But not here:

``` js
print('start 2');
hist(repeat(1e5, flip))
meh
print('end 2')
```

**Comment by longouyang** [comment]
This has something to do with side effects, as the error is caught here:

``` js
print('start 2');
repeat(1e4, flip)[0]
meh
print('end 2')
```

**Comment by longouyang** [comment]
Some insight.. on my machine, this works:

``` js
print('start 2');
hist(repeat(1e3, flip))
meh
print('end 2')
```

(repeat only `1e3` times instead of `1e5`)

I think what's happening is that error handling doesn't persist through trampoline pauses.

**Comment by longouyang** [comment]
Fixed in 75c4609
